.github/workflows/analysis_workflow.yml

```diff
@@ -0,0 +1,293 @@
+
+      - name: Extra envs
+        run: |
+          . build_tooling/vcpkg_caching.sh # Linux follower needs another call in CIBW
+          echo -e "VCPKG_BINARY_SOURCES=$VCPKG_BINARY_SOURCES
+          VCPKG_ROOT=$PLATFORM_VCPKG_ROOT" | tee -a $GITHUB_ENV
+          cmake -P cpp/CMake/CpuCount.cmake | sed 's/^-- //' | tee -a $GITHUB_ENV
+        env:
+          CMAKE_BUILD_PARALLEL_LEVEL: ${{vars.CMAKE_BUILD_PARALLEL_LEVEL}}
+
+      - name: Prepare C++ compilation env
+        run: . build_tooling/prep_cpp_build.sh 
+      
+      - name: CMake compile
+        uses: lukka/run-cmake@v10
+        with:
+          cmakeListsTxtPath: ${{github.workspace}}/cpp/CMakeLists.txt
+          configurePreset: ${{env.ARCTIC_CMAKE_PRESET}}
+          buildPreset: ${{env.ARCTIC_CMAKE_PRESET}}
+        env:
+          ARCTICDB_CODE_COVERAGE_BUILD: 1
+        
+      - name: Run C++ Tests
+        shell: bash -l {0}
+        run: |
+          cd cpp/out/linux-debug-build/
+          make -j ${{ steps.cpu-cores.outputs.count }} arcticdb_rapidcheck_tests
+          make -j ${{ steps.cpu-cores.outputs.count }} test_unit_arcticdb
+          ctest
+
+      - name: Set real storage variables
+        run: |
+          # Common
+          echo "ARCTICDB_PERSISTENT_STORAGE_TESTS=1" >> $GITHUB_ENV
+          echo "ARCTICDB_PERSISTENT_STORAGE_BRANCH_NAME=${{ github.run_id }}" >> $GITHUB_ENV
+          # S3 Specific
+          # TODO: Add these as GitHub variables
+          echo "ARCTICDB_REAL_S3_BUCKET=arcticdb-ci-test-bucket-01" >> $GITHUB_ENV
+          echo "ARCTICDB_REAL_S3_ENDPOINT=s3.eu-west-2.amazonaws.com" >> $GITHUB_ENV
+          echo "ARCTICDB_REAL_S3_REGION=eu-west-2" >> $GITHUB_ENV
+          echo "ARCTICDB_REAL_S3_CLEAR=1" >> $GITHUB_ENV
+          echo "ARCTICDB_REAL_S3_ACCESS_KEY=${{ secrets.AWS_S3_ACCESS_KEY }}" >> $GITHUB_ENV
+          echo "ARCTICDB_REAL_S3_SECRET_KEY=${{ secrets.AWS_S3_SECRET_KEY }}" >> $GITHUB_ENV
+
+      - name: Test with pytest
+        shell: bash -l {0}
+        run: |
+          yum install nodejs npm -y
+          npm config set strict-ssl false
+          npm install -g azurite
+          /opt/python/cp36-cp36m/bin/python -m pip install --upgrade pip
+          /opt/python/cp36-cp36m/bin/python -m pip install pycparser six cffi urllib3 python-dateutil PyJWT jmespath idna cryptography charset-normalizer certifi requests py pluggy path oauthlib MarkupSafe iniconfig dataclasses botocore attrs Werkzeug termcolor s3transfer requests-oauthlib pytz pytest protobuf portalocker path.py numpy msal mock Jinja2 itsdangerous isodate grpcio execnet decorator contextlib2 click azure-core xxhash xmltodict sortedcontainers retry responses pyyaml pytest-shutil pytest-fixture-config psutil prometheus-client pandas msrest msgpack msal-extensions grpcio-tools future flask colorama boto3 pytest-timeout pytest-server-fixtures pytest-cpp pymongo moto hypothesis flask-cors azure-storage-blob azure-identity coverage pytest-xdist
```

Comment:
This won't be maintainable when deps change. We have a similar problem elsewhere (shared an internal link with you). Guess we can do it that way.